boa/src/exec/mod.rs

```diff
@@ -14,6 +14,7 @@ pub trait Executable {
 pub(crate) enum InterpreterState {
     Executing,
     Return,
+    Error,
```

Comment:
Maybe we can allow the clippy lint that complains that this variant is never constructed, since it is only constructed with the `vm` feature on. Or maybe only compile it with the `vm` feature on?